src/components/combobox/combobox.tsx

```diff
@@ -386,6 +389,17 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
 
   private activeTransitionProp = "opacity";
 
+  private OpenCloseState: OpenCloseState = {
+    beforeOpen: () => this.calciteComboboxBeforeOpen.emit(this.publicEmitPayload),
+    open: () => this.calciteComboboxOpen.emit(this.publicEmitPayload),
+    beforeClose: () => this.calciteComboboxBeforeClose.emit(this.publicEmitPayload),
+    close: () => this.calciteComboboxClose.emit(this.publicEmitPayload)
+  };
+
+  private publicEmitPayload: any = {
+    element: this
+  };
+
```

Comment:
I split the function up to understand which parts are shared to be extracted and the payload seems to vary (eg. `combobox` only emits `el`, but `alert` also emits a `queue`).

Are we expecting components to vary on `OpenCloseState` as well (eg. would some components maybe have `close` but not the `open` event emitted)?

Reply:
I don't think we should make these be the same across components. It may depend but from what I see, we really don't need to emit anything. The `target/currentTarget` on the event should be the `el` and the user can get the open state from that.

Reply:
I feel like there was a PR asking for standardization in emitting a payload which involved ensuring there is always one emitted. But I can't seem to find it. I'll remove it from here and maybe add a discussion topic.